nsqd/diskqueue.go

```diff
@@ -261,6 +266,14 @@ func (d *diskQueue) readOne() ([]byte, error) {
 		return nil, err
 	}
 
+	if msgSize < d.minMsgSize || msgSize > d.maxMsgSize {
```

Comment:
I don't know if `msgSize > d.maxMsgSize` means it's _corrupt_?  It should _skip_ it, but does it need to discard the whole segment?